src/structures/channel/GuildChannel.ts

```diff
@@ -0,0 +1,32 @@
+import { APIChannel, APIOverwrite, Snowflake } from 'discord-api-types';
+import { Guild } from '../guild/Guild';
+import { Channel } from './Channel';
+
+export class GuildChannel extends Channel {
+	public name?: string;
+	public guildId?: Snowflake;
+	public position?: number;
+	public permissionOverwrites?: APIOverwrite[];
+	public nsfw!: boolean;
+	public parentId?: Snowflake;
+
+	public constructor(public readonly guild: Guild, data: APIChannel) {
+		super(guild.client, data);
+
+		this.$patch(data);
+	}
+
+	public $patch(data: APIChannel): void {
+		super.$patch(data);
+
+		if (data.parent_id) {
+			this.parentId = data.parent_id;
+		}
+
+		this.name = data.name;
+		this.guildId = data.guild_id;
+		this.position = data.position;
+		this.permissionOverwrites = data.permission_overwrites;
+		this.nsfw = !!data.nsfw;
```

Comment:
same as is_dirty

Reply:
This structure is there for the future methods that will be inside.